### DISCUSSION TITLE
Search/SEO friendly.

### DISCUSSION THREAD

**Discussion by Kpudlo** [community/community#3318] [Discussions] [answered]
Howdy, 

Are there any specific steps that are required to get Google to properly search the Discussions of a Repository? 

For example, the following site searches work, but we have been able to get this to work for our repo. 
Example 1: `site:https://github.com/github/feedback/discussions`
Example 2: `site:https://github.com/vercel/vercel/discussions/`

Thanks, 
Kevin

**Comment by PranamBhat** [comment]
Nothing as such!

**Comment by avital** [comment]
We are also hitting this on Flax. For example, I couldn't find https://github.com/google/flax/discussions/1081 via a Google search. This is very important for us as we're using GitHub discussions as a de-facto "source of truth" for best practices using Flax (sort of like Stack Overflow, but allowing us better control of the content")

**Comment by Kpudlo** [comment]
We found SEO started to work after adding the Discussions URL to the Repo's readme. The Discussions home showed up in SEO results rather quickly, but it took a while (2-3 weeks) for individual topics to start populating in SEO.

It works for now, but it's not the best solution.

**Comment by SEOGoddess** [comment, marked as answer]
> Are there any specific steps that are required to get Google to properly search the Discussions of a Repository?

It's pretty normal that it will take some time before you search results for new URLs/pages created (that includes new discussions). I would give it 90 days at least, especially when you make changes to a page's content. 

With that said, here's a bit of a checklist that will help Google to pick up the page:
1) Keyword focus - what keyword (a word or phrase) do you want this to rank for?
2) Keyword placement - that keywords should go on the page in the title and the main bod at least once, but not too many times. If you have 100 words for a quick discussion topic, then one mention is good. If you have 300 words or more then a couple mentions will work.
3) Link to the page - An external link from a reputable website helps give the page a boost. Google will think it's just another user generated page (which can tend to be spammy) and may not index or even rank it for that reason. A link from your website, a publication or a blog is a great way to get that page recognized quickly. 
Be sure to have come content on the page and the website that the link is on, and the the topic of that page matches the keyword you're targeting. 

If you can do those three things, you should be good to go.

**Comment by Alvao-1** [comment]
Making a site Web optimization well disposed implies that Google and other web search [[tools](https://toolshigh.com/bar-and-chain-oil-substitute/] can slither each page on the site proficiently, decipher the substance really, and list it in their data set.

**Comment by alexblackkot9** [comment]
Hello there! I know this thread is quite old, but I'd still like to share my 2 cents to revive it. First thing I would do if I were you would be to make sure I use relevant keywords in the titles and descriptions. This will help Google understand what discussions are about. Additionally, I'd try to write my content in a clear and easy-to-understand way. This will make it more likely that people will find my discussions helpful and share them with others. Btw, have you heard of LeadHarvestHQ? It's a great tool for generating leads and finding potential customers for your business. You can learn more about it at [leadharvesthq.com](https://leadharvesthq.com/). I hope this advice helps, even though this post is a old.

**Comment by musab-seo** [comment]
Hi Kevin,

There isn't a specific setting that guarantees Google will index GitHub Discussions. Public discussions can be crawled, but Google decides individually which pages are indexed and shown in search results. The site: operator can also be inconsistent, so not seeing a discussion there doesn't necessarily mean it hasn't been crawled.

For projects that publish useful web-based tools, keeping the important information accessible from the main repository can help with discoverability. For example, if a repository documents a construction-related tool such as a [concrete block calculator](https://concretecalculatorz.com/concrete-block-calculator), it can be helpful to explain the calculation method, inputs, and expected results in the repository or README rather than relying only on a Discussion page.

I'd also check the individual discussion URL in Google Search Console if you manage the repository. That can give you a better indication of whether Google has discovered and indexed the page than relying solely on a site: search.
